src/main/java/gyro/google/compute/AbstractAutoscalerResource.java

```diff
@@ -0,0 +1,242 @@
+        this.statusDetail = statusDetail;
+    }
+
+    @Override
+    public void copyFrom(Autoscaler model) {
+        setName(model.getName());
+        setAutoscalingPolicy(Optional.ofNullable(model.getAutoscalingPolicy())
+            .map(e -> {
+                ComputeAutoscalingPolicy computeAutoscalingPolicy = newSubresource(ComputeAutoscalingPolicy.class);
+                computeAutoscalingPolicy.copyFrom(e);
+                return computeAutoscalingPolicy;
+            })
+            .orElse(null));
+        setDescription(model.getDescription());
+        // `recommendedSize` is not available from Google SDK yet.
+        setRecommendedSize(Optional.ofNullable(model.get("recommendedSize"))
+            .filter(Integer.class::isInstance)
+            .map(Integer.class::cast)
+            .orElse(null));
+        setSelfLink(model.getSelfLink());
+        setStatus(model.getStatus());
+
+        List<ComputeAutoscalerStatusDetails> diffableAutoscalerStatusDetails = null;
+        List<AutoscalerStatusDetails> statusDetails = model.getStatusDetails();
+
+        if (statusDetails != null && !statusDetails.isEmpty()) {
+            diffableAutoscalerStatusDetails = statusDetails
+                .stream()
+                .map(e -> {
+                    ComputeAutoscalerStatusDetails computeVersion = newSubresource(
+                        ComputeAutoscalerStatusDetails.class);
+                    computeVersion.copyFrom(e);
+                    return computeVersion;
+                })
+                .collect(Collectors.toList());
+        }
+        setStatusDetail(diffableAutoscalerStatusDetails);
+    }
+
+    @Override
+    protected void doCreate(GyroUI ui, State state) throws Exception {
+        Autoscaler autoscaler = new Autoscaler();
+        autoscaler.setName(getName());
+        Optional.ofNullable(getAutoscalingPolicy())
+            .map(ComputeAutoscalingPolicy::copyTo)
+            .ifPresent(autoscaler::setAutoscalingPolicy);
+        autoscaler.setDescription(getDescription());
+
+        insert(autoscaler);
+        refresh();
+    }
+
+    @Override
+    protected void doUpdate(GyroUI ui, State state, Resource current, Set<String> changedFieldNames)
+        throws Exception {
+        Autoscaler autoscaler = new Autoscaler();
+        boolean shouldPatch = false;
+
+        for (String changedFieldName : changedFieldNames) {
+            if (changedFieldName.equals("description")) {
+                autoscaler.setDescription(getDescription());
+                shouldPatch = true;
+            } else if (changedFieldName.equals("autoscaling-policy")) {
+                ComputeAutoscalingPolicy diffableAutoscalingPolicy = getAutoscalingPolicy();
+                autoscaler.setAutoscalingPolicy(diffableAutoscalingPolicy == null
+                    ? Data.nullOf(AutoscalingPolicy.class)
+                    : diffableAutoscalingPolicy.copyTo());
+                shouldPatch = true;
+            }
+        }
```

Comment:
Is there a specific reason to use a `for` over here ?
Having the loop checks one of the condition twice in case both the fields changes.

Can be replaced by 
```
           if (changedFieldName.contains("description")) {
                autoscaler.setDescription(getDescription());
                shouldPatch = true;
            }
           
            if (changedFieldName.contains("autoscaling-policy")) {
                ComputeAutoscalingPolicy diffableAutoscalingPolicy = getAutoscalingPolicy();
                autoscaler.setAutoscalingPolicy(diffableAutoscalingPolicy == null
                    ? Data.nullOf(AutoscalingPolicy.class)
                    : diffableAutoscalingPolicy.copyTo());
                shouldPatch = true;
            }
```

Reply:
There's no obviously better solution as the suggested solution checks both conditions in case only one field changes, so best solution depends on the actual use cases.
Fixes have been provided even though I still think it's too much.